runelite-client/src/main/java/net/runelite/client/plugins/rememberusername/RememberUsernamePlugin.java

```diff
@@ -79,4 +80,14 @@ public void onGameStateChange(GameStateChanged event)
 			config.username(client.getUsername());
 		}
 	}
-}
+
+	@Subscribe
+	public void onSessionOpen(SessionOpen event)
+	{
```

Comment:
Instead of doing this with the new plugin manager changes, move this logic to startUp. And call client.setUsername("") on shutDown.

Reply:
Also add check for gamestate